zerver/views/message_edit.py

```diff
@@ -161,10 +161,13 @@ def update_message_backend(request: HttpRequest, user_profile: UserMessage,
     mention_user_ids: Set[int] = set()
     mention_data: Optional[MentionData] = None
     if content is not None:
+        leading_space = len(content) - len(content.lstrip(' '))
         content = content.strip()
         if content == "":
             content = "(deleted)"
         content = truncate_body(content)
+        if leading_space >= 4:
+            content = " " * leading_space + content
```

Comment:
We should just change `content.strip()` to `content.rstrip()` instead of trying to separately parse out the whitespace, remove the whitespace, and then add it back.

Reply:
agreed

Reply:
Thanks for the feedback!